src/array.jl

```diff
@@ -383,6 +383,13 @@ end
     @inbounds A.refs[I...] = get!(A.pool, v)
 end
 
+Base.fill(v::CategoricalValue{T}, dims::NTuple{N, Integer}) where {T, N} =
+    CategoricalArray{T, N}(fill(level(v), dims), copy(pool(v)))
```

Comment:
Nice - I understand that we invoke an inner constructor - right (like the discussion we had for `DataFrame` recently)?

Reply:
Yes.